tsdat/io/writers.py

```diff
@@ -170,54 +171,104 @@ def write(
+        header_df.to_csv(
+            filepath.with_suffix(".header.csv"),  # type: ignore
+            header=False,
+        )
+
+        # dimensions/coordinates
+        dim_dict = {}
+        for dim in dataset.dims:
+            dim_dict[dim] = dataset[dim].to_numpy()  # type: ignore
+        dim_df = pd.DataFrame.from_dict(dim_dict, orient="index")
+        dim_df.to_csv(
+            filepath.with_suffix(".dims.csv"),  # type: ignore
+            header=False,
+        )
+        # dimensions/coordinates
+
         for var in dataset:
-            shp = dataset[var].shape
-            if len(shp) <= 1:
-                d1.append(var)
-            elif len(shp) == 2:
-                d2.append(var)
+            # set attributes
+            attrs_dict = {}
+            attrs_dict["name"] = var
+            for att in dataset[var].attrs:
+                if att not in attributes:
+                    attributes.append(att)
+                attrs_dict[att] = dataset[var].attrs[att]
+            df_dictionary = pd.DataFrame([attrs_dict])
+            attribute_df = pd.concat([attribute_df, df_dictionary], ignore_index=True)  # type: ignore
+            # set attributes
+
+            dims = [coor for coor in dataset[var].coords]  # get dimensions of variable
+
+            # organize 1 dimension data
+            if len(dataset[var].to_numpy().shape) == 1 and len(dims) == 1:  # type: ignore
+                oneDdata[dims[0]] = dataset[var].coords[dims[0]].to_numpy()  # type: ignore
+                oneDdataCont[dims[0]][var] = dataset[var].to_numpy()  # type: ignore
+            # organize 1 dimension data
+
+            # organize 2 dimension data
+            elif (
+                len(dataset[var].to_numpy().shape) == 2  # type: ignore
+                and len(dims) == 2
+                and "time" in dims
+            ):
+                tempD = ""
+                for coord in dataset[var].coords:
+                    if coord != "time":
+                        twoDdata[coord] = dataset[var].coords[coord].to_numpy()  # type: ignore
+                        tempD = coord  # type: ignore
+                twoDdataCont[tempD][var] = dataset[var].to_numpy()  # type: ignore
+            # organize 2 dimension dat
+
+            # other dim data
             else:
-                warnings.warn(
-                    "CSV writer cannot save variables with more than 2 dimensions."
+                warnings.warn("Cannot convert " + var + " to .CSV")  # type: ignore
+            # other dim data
+
+        attribute_df.to_csv(
+            filepath.with_suffix(".attrs.csv"),  # type: ignore
+            index=False,
+        )
+
+        # convert organized 1 dimension data to CSV
+        for dim in oneDdataCont:  # type: ignore
+            one_dim_df = pd.DataFrame.from_dict(oneDdataCont[dim])  # type: ignore
+            one_dim_df.insert(0, dim, oneDdata[dim])  # type: ignore
```

Comment:
Seems like this is inserting a new column at the beginning for the dimension name?

If so then the interior of this loop could be a 1-liner:

`pd.DataFrame.from_dict({dim: data_1D[dim], **data_1D_cont}).to_csv(...))`

Also note that `one_dim_df` is never accessed after this loop